CONTRIBUTING.md

```diff
@@ -90,7 +90,7 @@ git push -f
 ```
 
 ### Merging pull requests
-Pull requests must be merged by a core maintainer using the `Merge pull request` option.
+Pull requests by default must be merged by a core maintainer using the `Merge pull request` option. They can be automatically merged by assigning the `automerge` label to the pull request, see [here](https://github.com/openservicemesh/osm/blob/main/docs/developer/automerge.md) for additional details.
```

Comment:
We should document who is expected to assign the label. Maybe it makes the most sense to follow the same pattern we have for manually merging PRs: If the author is a maintainer, then they assign the label. If the author is an external contributor, then any maintainer can assign the label.